objects/src/mock/chain.rs

```diff
@@ -55,3 +55,248 @@ pub fn mock_chain_data(consumed_notes: &mut [Note]) -> ChainMmr {
+
+            // TODO: Set batch_root and proof_hash to the correct values once the kernel is
+            // available.
+            let batch_root = Digest::default();
+            let proof_hash = Digest::default();
+
+            let new = BlockHeader::new(
+                prev_hash,
+                block_num,
+                chain_root,
+                Default::default(),
+                nullifier_root,
+                Default::default(),
+                batch_root,
+                proof_hash,
+                version,
+                timestamp,
+            );
+
+            self.blocks.push(new);
+            self.chain.mmr_mut().add(new.hash());
+            self.objects.extend(&mut self.pending_objects);
+
+            new
+        }
+
+        /// Creates a [Account] and add to the list of pending objects.
+        pub fn new_account<C, S, A>(
+            &mut self,
+            code: C,
+            storage: S,
+            assets: A,
+            immutable: bool,
+            on_chain: bool,
+        ) -> Result<AccountId, AccountError>
+        where
+            C: AsRef<str>,
+            S: IntoIterator<Item = StorageItem>,
+            A: IntoIterator<Item = Asset>,
+        {
+            let account_type = if immutable {
+                AccountType::RegularAccountImmutableCode
+            } else {
+                AccountType::RegularAccountUpdatableCode
+            };
+
+            let rng = R::from_rng(&mut self.rng).expect("rng seeding failed");
+            let account = AccountBuilder::new(rng)
+                .add_storage_items(storage)
+                .add_assets(assets)
+                .account_type(account_type)
+                .on_chain(on_chain)
+                .code(code)
+                .build()?;
+            let account_id = account.id();
+            self.pending_objects.accounts.push(account);
+            Ok(account_id)
+        }
+
+        /// Creates a [AccountId] with type [AccountType::FungibleFaucet] and add to the list of
+        /// pending objects.
+        pub fn new_fungible_faucet(
+            &mut self,
+            on_chain: bool,
+            code_root: Digest,
+            storage_root: Digest,
+        ) -> Result<AccountId, AccountError> {
+            let faucet_id = self
+                .account_id_builder
+                .account_type(AccountType::NonFungibleFaucet)
+                .on_chain(on_chain)
+                .code_root(code_root)
+                .storage_root(storage_root)
+                .build()?;
+            let builder = FungibleAssetBuilder::new(faucet_id)
+                .expect("builder was not configured to create fungible faucets");
+            self.pending_objects.fungible_faucets.push((faucet_id, builder));
+            Ok(faucet_id)
+        }
+
+        /// Creates a [AccountId] with type [AccountType::NonFungibleFaucet] and add to the list of
+        /// pending objects.
+        pub fn new_nonfungible_account(
```

Comment:
nit: should this be `new_nonfungible_faucet`?

Reply:
changed